src/util/map.js

```diff
@@ -146,3 +146,9 @@ export function merge<K, V>(
   }
   return result;
 }
+
+export function pushValue<K, V>(map: Map<K, V[]>, key: K, value: V): void {
+  const arr = map.get(key) || [];
+  arr.push(value);
+  map.set(key, arr);
```

Comment:
Since this is factored out into a utility function, we might as well use
the twice-as-fast-in-the-usual-case implementation:

```js
let arr = map.get(key);
if (arr == null) {
  map.set(key, arr = []);
}
arr.push(value);
```

Or am I missing something?

Reply:
Done